azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusExtractor.java

```diff
@@ -568,33 +568,60 @@ private void logAppContainerStatuses() {
    * @return
    */
   public AzPodStatus createAzPodStatus() {
+    // All pod statuses are checked for any status extracted from the pod event. If multiple
+    // statuses are detected, the first derived pod status will be returned, and the conflict
+    // will be logged
+    int statusCount = 0;
+    AzPodStatus azPodStatus = AzPodStatus.AZ_POD_UNEXPECTED;
     if (checkForAzPodRequested()) {
-      return AzPodStatus.AZ_POD_REQUESTED;
+      azPodStatus = AzPodStatus.AZ_POD_REQUESTED;
+      statusCount++;
     }
-    logInitContainerStatuses();
     if (checkForAzPodScheduled()) {
-      return AzPodStatus.AZ_POD_SCHEDULED;
+      azPodStatus = updateAzPodStatus(statusCount++, azPodStatus, AzPodStatus.AZ_POD_SCHEDULED);
+    }
+    if (AzPodStatus.AZ_POD_REQUESTED != azPodStatus) {
```

Comment:
Please add comments for any condition evaluation.

Reply:
removed unnecessary if checks